scheduler/schedule.go

```diff
@@ -0,0 +1,93 @@
+package scheduler
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/axsh/openvdc/handlers"
+	"github.com/axsh/openvdc/model"
+)
+
+var (
+	// Time to wait for the offer to arrive
+	waitOfferSec = 5
+
+	instanceSchedulerHandlers = make(map[string]handlers.InstanceScheduleHandler)
+	schedule                  *Schedule
+)
+
+func init() {
+	schedule = newSchedule()
+}
+
+func RegisterInstanceScheduleHandler(name string, i handlers.InstanceScheduleHandler) error {
+	if _, exists := instanceSchedulerHandlers[name]; exists {
+		return fmt.Errorf("Duplicated name for instance schedule handler: %s", name)
+	}
+	instanceSchedulerHandlers[name] = i
+	return nil
+}
+
+type Schedule struct {
+	*sync.Mutex
```

Comment:
No need to expose Lock()/Unlock() methods. And also ``sync.RWMutex`` is better .

https://golang.org/pkg/sync/#RWMutex



```
m sync.RWMutex
```

Reply:
Modified mutex from sync.Mutex to sync.RWMutex and turn to private variable.